features/logging/elasticsearch.feature

```diff
@@ -86,3 +86,50 @@ Feature: Elasticsearch related tests
       | infra-00  |
       | .kibana_  |
       | .security |
+
+
+  # @author anli@redhat.com
+  # @case_id OCP-36368
+  @admin
+  @destructive
+  Scenario: elasticsearch nodes can scale down
+    And default storageclass is stored in the :default_sc clipboard
+    Given I delete the clusterlogging instance
+    Then the step should succeed
+    Given I use the "openshift-logging" project
+    And I register clean-up steps:
+    """
+    Given I delete the clusterlogging instance
+    Then the step should succeed
+    And I run the :delete client command with:
+      | object_type | pvc               |
+      | all         | true              |
+      | n           | openshift-logging |
+    Then the step should succeed
+    """
+    Given I obtain test data file "logging/clusterlogging/clusterlogging-storage-template.yaml"
+    When I process and create:
+      | f | clusterlogging-storage-template.yaml |
+      | p | ES_NODE_COUNT=3                      |
+      | p | STORAGE_CLASS=<%= cb.default_sc.name %>                                                      |
+      | p | PVC_SIZE=10Gi                                                                                |
+    Then the step should succeed
+    Given I wait for the "elasticsearch" elasticsearches to appear
+    Given I wait until ES cluster is ready
+    And the expression should be true> cluster_logging('instance').logstore_node_count == 3
+    And the expression should be true> elasticsearch('elasticsearch').nodes[0]['nodeCount'] == 3
+    When I run the :patch client command with:
+      | resource      | clusterlogging                                          |
+      | resource_name | instance                                                |
+      | p             | {"spec":{"logStore":{"elasticsearch":{"nodeCount":2}}}} |
+      | type          | merge                                                   |
+    Then the step should succeed
+    And the expression should be true> cluster_logging('instance').logstore_node_count == 2
+    Given I wait up to 600 seconds for the steps to pass:
+    """
+    And the expression should be true> elasticsearch('elasticsearch').nodes[0]['nodeCount'] == 2
```

Comment:
@anpingli I think we need add some other checkpoint here, such as: the ES pods count equals to 2. In some condition, the node count in elasticsearch/elasticsearch is changed, but no ES pod is deleted.

You can try something like:
```
And the expression should be true> BushSlicer::Pod.get_labeled("component=elasticsearch", project: project("openshift-logging", switch: false), user: admin).count == 2
```